ludwig/features/set_feature.py

```diff
@@ -169,11 +184,10 @@ class SetOutputFeature(SetFeatureMixin, OutputFeature):
     default_validation_metric = JACCARD
 
     def __init__(self, feature, output_features: Dict[str, OutputFeature]):
-        super().__init__(feature, output_features)
-
         self.num_classes = 0
         self.threshold = 0.5
 
+        super().__init__(feature, output_features)
```

Comment:
This is because we create the PredictModule (which uses these values) in the parent class' constructor. So if these values aren't set before then, it will blow up.

Reply:
Ah, I see. `create_predict_module()` is called by the super constructor, with `self.threshold`. I suppose `self.num_classes` could technically be initialized after the call to the super constructor, but it's not crucial.